rclcpp/test/rclcpp/test_parameter_client.cpp

```diff
@@ -291,3 +291,43 @@ TEST_F(TestParameterClient, sync_parameter_get_parameter_types) {
   ASSERT_EQ(1u, parameter_types.size());
   ASSERT_EQ(rclcpp::ParameterType::PARAMETER_INTEGER, parameter_types[0]);
 }
+
+/*
+  Coverage for async describe_parameters
+ */
+TEST_F(TestParameterClient, async_parameter_describe_parameters) {
+  auto asynchronous_client = std::make_shared<rclcpp::AsyncParametersClient>(node);
+  bool callback_called = false;
+  auto callback = [&callback_called](
+    std::shared_future<std::vector<rcl_interfaces::msg::ParameterDescriptor>> result)
+    {
+      // We expect the result to be empty since we tried to get a parameter that didn't exist.
```

Comment:
Can we also add a test here to check when some parameters *do* exist?  That would cover more of the code, I think.

Also, what happens if some of the parameters exist, and some don't?  It might be worth adding a test for that as well.